### PR TITLE
Sell Books Feature

### PR DESCRIPTION
**Pull request by mdvb1001**
@Zelig880 Here's a summary of the work in this PR:
- Allowed price to be saved with 2 decimal points
- Got the price to display in the owner's library (see pic 1)
- Got the price to display in another user's search results (see pic 2)
- Left a comment in the code that I would love your advice on 
- Are there any tests I can write for this? (I didn't find any test files for other backend attributes such as this one) 

image

image

### CODE REVIEW

File: resources/js/pages/library/view.vue [hunk 1]
```diff
@@ -64,6 +64,7 @@
                 :condition="conditions[result.condition].name"
                 :transaction-type="transactionType[result.type].nameForCustomer"
                 :distance="result.bookshelf.distance"
+                :price="parseFloat(result.price)"
```
Comment (mdvb1001) [inline resources/js/pages/library/view.vue:67]: @Zelig880 I don't love that I converted this to a float here. From what I can see, we are saving the price as a `Number`, so I'm not sure why it's coming back as `String` from the DB. Any ideas or thoughts on this?

File: resources/js/components/sections/bookshelfAddStep3.vue [hunk 2]
```diff
@@ -3,6 +3,12 @@
     <h3 class="text-2xl font-semibold mb-3"> {{ $t( 'bookshelfAdd-step3-title' ) }} </h3>
     <div class="grid grid-cols-1 md:grid-cols-2 gap-5 ml-8">
       <Button v-for="(Type, index) in transactionType" :key="Type.name" @click="selectType(index)" theme="square" :color="(selectedTypeIndex === index) ? 'primary' : ''">{{ Type.name }} </Button>
+      <div class="flex" v-if="selectedTypeIndex === 2">
+        <label>
+          {{ $t( 'bookshelfAdd-step3-price' ) }}
+        </label>
+        <input type="number" step="0.01" :value="price" class="BookshelfAddBookshelf-field rounded-md border-2 pl-8 pr-6 py-2 block mb-4 mt-1" @keyup="$emit('update:price', $event.target.value)" @change="$emit('update:price', $event.target.value)">
```
Comment (Zelig880) [inline resources/js/components/sections/bookshelfAddStep3.vue:10]: You should be able to use `v-model` here.. so remove :value and @keyup and @change and just add

```<input v-model="price" plus all the classes and the rest />```

The above will automatically get the value and set the value

File: resources/js/pages/library/view.vue (hunk 1 above)
Comment (Zelig880) [reply resources/js/pages/library/view.vue:67]: I will have to investigate this. It is probably a miss configuration of the db schema or the way the data is retrieved from the PHP side. Nothing to worry about for now, but good to note

**Review by Zelig880 — CHANGES_REQUESTED** [review]
Just one small change to clean up one of the event emitted.

File: resources/js/components/sections/bookshelfAddStep3.vue (hunk 2 above)
Comment (mdvb1001) [reply resources/js/components/sections/bookshelfAddStep3.vue:10]: I tried implementing this, but the price isn't updating as it should. I will investigate a bit later.

**Comment by mdvb1001** [comment]
@Zelig880 This is ready for you now, Simone :)

**Review by Zelig880 — APPROVED** [review]

**Pull request merged**